src/utils/utils.cpp

```diff
@@ -582,4 +586,31 @@ bool octalText2hexText (const string &octalText, string &hexText)
     } while (true);
     return true;
 }
-*/
\ No newline at end of file
+*/
+
+// Get hostname and IP address
+void getNetworkInfo (string &hostname, string &ipAddress)
+{
+    char host[256];
+    char *ip;
+    struct hostent *host_entry;
+    int iResult;
+    iResult = gethostname(host, sizeof(host));
+    if (iResult == -1)
+    {
+        cerr << "Error: getNetworkInfo() failed calling gethostname()" << endl;
+        hostname = "(error calling gethostname)";
+        ipAddress = "(error calling gethostname)";
+        return;
+    }
+    hostname = host;
+    host_entry = gethostbyname(host);
```

Comment:
gethostbyname isn't thread safe and it's obsolete.
man(description):  The gethostbyname*(), gethostbyaddr*(), herror(), and hstrerror() functions are obsolete.  Applications should use getaddrinfo(3), getnameinfo(3), and gai_strerror(3) instead.

Reply:
To be done in a  separate PR